### ISSUE TITLE
Install fails on Raspberry Pi 2 B+

### ISSUE THREAD

**Issue by NiveditN** [peterbraden/node-opencv#275]
I installed OpenCV from the Raspbian repository using the following command which should give me OpenCV 2.4.1 along with python:

```
sudo apt-get install libopencv-dev python-opencv
```
Then I attempted to install node-opencv several times  but each time it failed with the following error:
```
code ELIFELCYLE
```
```
node-pre-gyp install --fallback-to-build
```
I also installed node-pre-gyp and node-gyp separately and tried again. Still the same error occurs.

The install log is attached below:
```
sudo npm install -g opencv
npm WARN installMany node-pre-gyp was bundled with opencv@3.0.0, but bundled package wasn't found in unpacked tree

> opencv@3.0.0 install /usr/local/lib/node_modules/opencv
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/0.12.6"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/opencv/.node-gyp"
make: Entering directory '/usr/local/lib/node_modules/opencv/build'
  CXX(target) Release/obj.target/opencv/src/init.o
  CXX(target) Release/obj.target/opencv/src/Matrix.o
  CXX(target) Release/obj.target/opencv/src/OpenCV.o
  CXX(target) Release/obj.target/opencv/src/CascadeClassifierWrap.o
  CXX(target) Release/obj.target/opencv/src/Contours.o
  CXX(target) Release/obj.target/opencv/src/Point.o
  CXX(target) Release/obj.target/opencv/src/VideoCaptureWrap.o
  CXX(target) Release/obj.target/opencv/src/CamShift.o
  CXX(target) Release/obj.target/opencv/src/HighGUI.o
  CXX(target) Release/obj.target/opencv/src/FaceRecognizer.o
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::New(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:65:80: error: too many arguments to function ‘cv::Ptr<cv::FaceRecognizer> cv::createLBPHFaceRecognizer(int, int, int, int)’
/usr/include/opencv2/contrib/contrib.hpp:951:36: note: declared here
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::CreateLBPH(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:91:3: error: too many arguments to function ‘cv::Ptr<cv::FaceRecognizer> cv::createLBPHFaceRecognizer(int, int, int, int)’
/usr/include/opencv2/contrib/contrib.hpp:951:36: note: declared here
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::CreateEigen(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:111:3: error: too many arguments to function ‘cv::Ptr<cv::FaceRecognizer> cv::createEigenFaceRecognizer(int)’
/usr/include/opencv2/contrib/contrib.hpp:949:36: note: declared here
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::CreateFisher(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:131:3: error: too many arguments to function ‘cv::Ptr<cv::FaceRecognizer> cv::createFisherFaceRecognizer(int)’
/usr/include/opencv2/contrib/contrib.hpp:950:36: note: declared here
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::UpdateSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:220:14: error: ‘class cv::FaceRecognizer’ has no member named ‘update’
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::PredictSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:235:52: error: no matching function for call to ‘cv::FaceRecognizer::predict(cv::Mat&, int&, double&)’
../src/FaceRecognizer.cc:235:52: note: candidate is:
/usr/include/opencv2/contrib/contrib.hpp:931:21: note: virtual int cv::FaceRecognizer::predict(cv::InputArray) const
/usr/include/opencv2/contrib/contrib.hpp:931:21: note:   candidate expects 1 argument, 3 provided
../src/FaceRecognizer.cc: In static member function ‘static void FaceRecognizerWrap::GetMat(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/FaceRecognizer.cc:271:26: error: ‘class cv::FaceRecognizer’ has no member named ‘getMat’
opencv.target.mk:100: recipe for target 'Release/obj.target/opencv/src/FaceRecognizer.o' failed
make: *** [Release/obj.target/opencv/src/FaceRecognizer.o] Error 1
make: Leaving directory '/usr/local/lib/node_modules/opencv/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Linux 3.18.14-v7+
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=opencv" "--configuration=Release" "--module_name=opencv" "--version=3.0.0" "--major=3" "--runtime=node" "--node_abi=node-v14" "--platform=linux" "--target_platform=linux" "--arch=arm" "--target_arch=arm" "--module_main=./lib/opencv" "--host=https://node-opencv.s3.amazonaws.com/" "--module_path=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm" "--module=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm/opencv.node" "--remote_path=./opencv/v3.0.0/Release/" "--package_name=node-v14-linux-arm.tar.gz" "--staged_tarball=build/stage/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz" "--hosted_path=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/" "--hosted_tarball=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz"
gyp ERR! cwd /usr/local/lib/node_modules/opencv
gyp ERR! node -v v0.12.6
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/node-gyp/bin/node-gyp.js rebuild --name=opencv --configuration=Release --module_name=opencv --version=3.0.0 --major=3 --runtime=node --node_abi=node-v14 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./lib/opencv --host=https://node-opencv.s3.amazonaws.com/ --module_path=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm --module=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm/opencv.node --remote_path=./opencv/v3.0.0/Release/ --package_name=node-v14-linux-arm.tar.gz --staged_tarball=build/stage/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz --hosted_path=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/ --hosted_tarball=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/opencv/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:1015:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
node-pre-gyp ERR! System Linux 3.18.14-v7+
node-pre-gyp ERR! command "node" "/usr/local/lib/node_modules/opencv/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/local/lib/node_modules/opencv
node-pre-gyp ERR! node -v v0.12.6
node-pre-gyp ERR! node-pre-gyp -v v0.5.31
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/node-gyp/bin/node-gyp.js rebuild --name=opencv --configuration=Release --module_name=opencv --version=3.0.0 --major=3 --runtime=node --node_abi=node-v14 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./lib/opencv --host=https://node-opencv.s3.amazonaws.com/ --module_path=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm --module=/usr/local/lib/node_modules/opencv/build/opencv/v3.0.0/Release/node-v14-linux-arm/opencv.node --remote_path=./opencv/v3.0.0/Release/ --package_name=node-v14-linux-arm.tar.gz --staged_tarball=build/stage/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz --hosted_path=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/ --hosted_tarball=https://node-opencv.s3.amazonaws.com/opencv/v3.0.0/Release/node-v14-linux-arm.tar.gz' (1)
npm ERR! Linux 3.18.14-v7+
npm ERR! argv "node" "/usr/local/bin/npm" "install" "-g" "opencv"
npm ERR! node v0.12.6
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! opencv@3.0.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the opencv@3.0.0 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the opencv package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls opencv
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pi/npm-debug.log
```

**Comment by funky81** [comment]
i'm having the same problems....have you figure how to do it?

**Comment by NiveditN** [comment]
I reinstalled version 2.4.10 as shown in this [guide](http://www.pyimagesearch.com/2015/02/23/install-opencv-and-python-on-your-raspberry-pi-2-and-b/). It's longer, and the build process will take some 3 hours on RPi 2, but at least it makes node-opencv work.

**Comment by peterbraden** [comment]
Fixed in cbaff9a
